src/plugins/crypto/test_internals.h

```diff
@@ -17,68 +18,24 @@
 #include <tests_internal.h>
 #include <tests_plugin.h>
 
-/*
- * The test vectors are taken from NIST SP 800-38A, section F.2.5 "CBC-AES256.Encrypt"
- * See <http://csrc.nist.gov/publications/nistpubs/800-38a/sp800-38a.pdf> for further information.
- */
-static const unsigned char key[] = { 0x60, 0x3d, 0xeb, 0x10, 0x15, 0xca, 0x71, 0xbe, 0x2b, 0x73, 0xae, 0xf0, 0x85, 0x7d, 0x77, 0x81,
-				     0x1f, 0x35, 0x2c, 0x07, 0x3b, 0x61, 0x08, 0xd7, 0x2d, 0x98, 0x10, 0xa3, 0x09, 0x14, 0xdf, 0xf4 };
+#include "test_key.h"
 
-static const unsigned char iv[] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f };
+#define TEST_KEY_ID "DDEBEF9EE2DC931701338212DAF635B17F230E8D"
 
+#define MIN(a, b) (((a) < (b)) ? (a) : (b))
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
 
 static const char strVal[] = "abcde";
 static const kdb_octet_t binVal[] = { 0x01, 0x02, 0x03, 0x04 };
 
-/**
- * @brief create new KeySet and add a working configuration to it.
- * @param shutdown set to 1 if the shutdown key should be appended.
- *                 If this key is set, the crypto library will be de cleaned up.
- */
-static KeySet * newWorkingConfiguration (int shutdown)
+static int isMarkedForEncryption (const Key * k)
 {
-	Key * configKey = keyNew ("user/crypto/key", KEY_END);
-	keySetBinary (configKey, key, sizeof (key));
-
-	Key * configIv = keyNew ("user/crypto/iv", KEY_END);
-	keySetBinary (configIv, iv, sizeof (iv));
-
-	if (shutdown)
+	const Key * metaEncrypt = keyGetMeta (k, ELEKTRA_CRYPTO_META_ENCRYPT);
+	if (metaEncrypt && strcmp (keyString (metaEncrypt), "1") == 0)
 	{
-		Key * configShutdown = keyNew ("user/shutdown", KEY_END);
-		keySetString (configShutdown, "1");
-		return ksNew (3, configKey, configIv, configShutdown, KS_END);
+		return 1;
 	}
-
-	return ksNew (2, configKey, configIv, KS_END);
-}
-
-/**
- * @brief create new KeySet and add an invalid configuration to it.
- *
- * The cryptographic key in the returned KeySet has an invalid size.
- */
-static KeySet * newInvalidConfiguration ()
```

Comment:
The only possible invalid configuration is one without a GPG recipient key specified. I added a test case for that. Everything else is now handled with default values.